linera-core/src/client/mod.rs

```diff
@@ -1314,6 +1314,9 @@ impl<Env: Environment> Client<Env> {
                             "Protected incoming message failed to execute locally"
                         )
                     );
+                    if message.action == MessageAction::Reject {
```

Comment:
Rejecting a message should never fail. So an `ensure!` would be consistent.

Reply:
`ensure!` is just a bit awkward in this case: we'd want to return the error that we know the `result` is. So I'd prefer just returning that over having `result.unwrap_err()` inside `ensure!`.

Or do you think it should be a different error?

I'd prefer to leave it like this on `testnet_conway`, because as far as the code goes, it's not unreachable. It's only unexpected once all chains have migrated to an epoch with `FLAG_FREE_REJECT`.